metabase/metabase
# fix(sdk): reduce visual artifacts on PDF/PNG exports on custom sdk themes

Partially fixes https://github.com/metabase/metabase/issues/48716

### Description

We support custom themes in the embedding sdk. Currently, PDF exports via "Export to PDF" on dashboards look broken on the sdk when a custom theme is applied. This PR partially fixes some of the visual artifacts present in the PDF exports, but not all of them.

### Fixes

- Background color of the export is incorrect. It does not follow the theme.
- Text are being cut off vertically.

### How to verify

- Run the Shoppy demo app locally
- Go to a dashboard, such as [/admin/analytics/17](https://embedded-analytics-sdk-demo.metabase.com/admin/analytics/17). Note that PDF exports are currently disabled, so you have to change `withDownloads={false}` to true in Shoppy in `DashboardPage.tsx`.
- Use the "Export as PDF" button on the top right, and the PNG exports in the dashcards.

Check that we are not breaking the main Metabase app (courtesy of Nicolo):

- Public and static embeds has three themes.
- Dashboards has the dark theme switch when in fullscreen
- There are dashboard downloads as PDF, cards on dashboard downloads as PNG, and standalone question downloads as PNG.

### Demo

Before - current state of PDF/PNG exports in Shoppy

![CleanShot 2567-10-16 at 02 17 03@2x](https://github.com/user-attachments/assets/5724e12f-304a-4a98-8cb7-c82055b539ca)

After apply the fix. Note that there are [more visual artifacts remaining](https://github.com/metabase/metabase/issues/48716) to be fixed later.

![CleanShot 2567-10-16 at 02 14 24@2x](https://github.com/user-attachments/assets/8e79b493-9032-4c3e-a380-60b6c7f7835f)

Static embedding still looks correct in the preview

![CleanShot 2567-10-16 at 18 05 17@2x](https://github.com/user-attachments/assets/5bfa7522-3003-4a13-add5-ae9c017a7c07)

![CleanShot 2567-10-16 at 18 05 12@2x](https://github.com/user-attachments/assets/f9d8d707-9103-49da-8bb9-64e6ced0f8ad)
